adapter-cqld4/src/main/java/io/nosqlbench/adapter/cqld4/opdispensers/CQLD4PreparedStmtDiagnostics.java

```diff
@@ -50,84 +52,82 @@
-    private final static Logger logger = LogManager.getLogger(CQLD4PreparedStmtDiagnostics.class);
+    private static final Logger logger = LogManager.getLogger(CQLD4PreparedStmtDiagnostics.class);
 
-    public static BoundStatement bindStatement(BoundStatement bound, CqlIdentifier colname, Object colval, DataType coltype) {
+    public static BoundStatement bindStatement(BoundStatement bound, CqlIdentifier colname,
+                                               Object colval, DataType coltype) {
 
-//        if (coltype instanceof PrimitiveType pt) {
-        try {
-            BoundStatement reproduce_error_with_custom_dataType = switch (coltype.getProtocolCode()) {
-                case CUSTOM -> throw new OpConfigError("Error with Custom DataType");
-                case ASCII, VARCHAR -> bound.setString(colname, (String) colval);
-                case BIGINT, COUNTER -> bound.setLong(colname, (long) colval);
-                case BLOB -> bound.setByteBuffer(colname, (ByteBuffer) colval);
-                case BOOLEAN -> bound.setBoolean(colname, (boolean) colval);
-                case DECIMAL -> bound.setBigDecimal(colname, (BigDecimal) colval);
-                case DOUBLE -> bound.setDouble(colname, (double) colval);
-                case FLOAT -> bound.setFloat(colname, (float) colval);
-                case INT -> bound.setInt(colname, (int) colval);
-                case SMALLINT -> bound.setShort(colname, (short) colval);
-                case TINYINT -> bound.setByte(colname, (byte) colval);
-                case TIMESTAMP -> bound.setInstant(colname, (Instant) colval);
-                case TIMEUUID, UUID -> bound.setUuid(colname, (UUID) colval);
-                case VARINT -> bound.setBigInteger(colname, (BigInteger) colval);
-                case INET -> bound.setInetAddress(colname, (InetAddress) colval);
-                case DATE -> bound.setLocalDate(colname, (LocalDate) colval);
-                case TIME -> bound.setLocalTime(colname, (LocalTime) colval);
-                case DURATION -> bound.setCqlDuration(colname, (CqlDuration) colval);
-                case LIST -> bound.setList(colname, (List) colval, ((List) colval).get(0).getClass());
-                case MAP -> {
-                    Map map = (Map) colval;
-                    Set<Map.Entry> entries = map.entrySet();
-                    Optional<Map.Entry> first = entries.stream().findFirst();
-                    if (first.isPresent()) {
-                        yield bound.setMap(colname, map, first.get().getKey().getClass(), first.get().getValue().getClass());
-                    } else {
-                        yield bound.setMap(colname, map, Object.class, Object.class);
-                    }
+        return switch (coltype.getProtocolCode()) {
+            case CUSTOM -> {
+                if (coltype instanceof CqlVectorType) {
+                    yield bound.setCqlVector(colname, (CqlVector<?>) colval);
```

Comment:
Nice add. It would be nice to come back to this for any other packaged Custom types.